dali/imgcodec/decoders/nvjpeg2k/nvjpeg2k.cc

```diff
@@ -89,7 +97,7 @@ NvJpeg2000DecoderInstance::~NvJpeg2000DecoderInstance() {
       res.intermediate_buffer.free();
     });
 
-  for (auto thread_id : tp_->GetThreadIds())
+  for (auto& thread_id : tp_->GetThreadIds())
```

Comment:
https://en.cppreference.com/w/cpp/thread/thread/id
```The class thread::id is a lightweight, trivially copyable class that serves as a unique identifier of std::thread objects.```
I think we are good with copy here

Reply:
:+1: